File: python/rna-transcription/rna_transcription.py
def to_rna(dnaStrand):
    rnaStrand = ""

    for nuc in dnaStrand:
        if (nuc == "G"):
            rnaStrand = rnaStrand+"C"
        elif (nuc == "C"):
            rnaStrand = rnaStrand+"G"
        elif (nuc == "T"):
            rnaStrand = rnaStrand+"A"
        elif (nuc == "A"):
            rnaStrand = rnaStrand+"U"

    if(len(rnaStrand) != len(dnaStrand)):
        rnaStrand = ""
        print("Invalid DNA strand input.")

    return rnaStrand

File: java/rna-transcription/src/main/java/RnaTranscription.java
class RnaTranscription
{

    String transcribe(String dnaStrand)
    {
      int len = dnaStrand.length();
      String rnaStrand = "";

      for (int i=0; i < len; i++)
      {
        switch (dnaStrand.charAt(i))
        {
          case 'G':
            rnaStrand += "C";
            break;
          case 'C':
            rnaStrand += "G";
            break;
          case 'T':
            rnaStrand += "A";
            break;
          case 'A':
            rnaStrand += "U";
            break;
          default:
            throw new IllegalArgumentException("Invalid input");
        }
      }
      return rnaStrand;
    }
}
